### ISSUE TITLE
compile with -Wall -Werror

### ISSUE THREAD

**Issue by chessai** [andrewthad/primitive-containers#19]
there are warnings about unmatched patterns that are quite worrying to me.

additionally, unused arguments to functions. these could be indicative of oversights that lead to bugs.

i'm of the opinion that code that makes heavy use of primitive should compile with -Wall -Werror, since it's so easy to make one-off errors like that.

**Comment by andrewthad** [comment]
I agree that this should be able to compile with `-Wall -Werror`. Any PRs helping fix this are welcome.

**Comment by chessai** [comment]
one problem with me creating a PR for the -Wall -Werror stuff is that it's in places where the logic is very dense, and function implementations are nontrivial. So it would take me considerably longer than you (the writer) to look at an unused argument and go through the motions of:

* Should this actually be used?
* If so, how?
* If it's not used, remove it, and fix any functions that may have called it.

I'm not volunteering you per se, just remarking that if I do get around to this, it will take me a while.
